src/simulator_core/entities/assets/demand_cluster.py

```diff
@@ -214,37 +214,29 @@ def write_to_output(self) -> None:
         The output list is a list of dictionaries, where each dictionary
         represents the output of its asset for a specific timestep.
         """
+        if not self.simulation_performed():
+            raise ValueError("Simulation data not available.")
         outputs = dict()
 
-        if self.simulation_performed():
-            outputs[PROPERTY_TEMPERATURE_SUPPLY] = self.pandapipes_net.res_flow_control['t_from_k'][
-                self._flow_control.index]
-            outputs[PROPERTY_TEMPERATURE_RETURN] = self.pandapipes_net.res_heat_exchanger['t_to_k'][
-                self._heat_exchanger.index]
-            outputs[PROPERTY_PRESSURE_SUPPLY] = self.pandapipes_net.res_flow_control['p_from_bar'][
-                self._flow_control.index]
-            outputs[PROPERTY_PRESSURE_RETURN] = self.pandapipes_net.res_heat_exchanger['p_to_bar'][
+        outputs[PROPERTY_TEMPERATURE_SUPPLY] = self.pandapipes_net.res_flow_control['t_from_k'][
+            self._flow_control.index]
+        outputs[PROPERTY_TEMPERATURE_RETURN] = self.pandapipes_net.res_heat_exchanger['t_to_k'][
+            self._heat_exchanger.index]
+        outputs[PROPERTY_PRESSURE_SUPPLY] = self.pandapipes_net.res_flow_control['p_from_bar'][
+            self._flow_control.index]
+        outputs[PROPERTY_PRESSURE_RETURN] = self.pandapipes_net.res_heat_exchanger['p_to_bar'][
+            self._heat_exchanger.index]
+        outputs[PROPERTY_MASSFLOW] = \
+            self.pandapipes_net.res_heat_exchanger['mdot_from_kg_per_s'][
                 self._heat_exchanger.index]
-            outputs[PROPERTY_MASSFLOW] = \
-                self.pandapipes_net.res_heat_exchanger['mdot_from_kg_per_s'][
-                    self._heat_exchanger.index]
-            outputs[PROPERTY_VOLUMEFLOW] = mass_flow_to_volume_flow(
-                outputs[PROPERTY_MASSFLOW],
-                outputs[PROPERTY_TEMPERATURE_SUPPLY],
-                self.pandapipes_net)
-            outputs[PROPERTY_THERMAL_POWER] = mass_flow_and_temperature_to_heat_demand(
-                outputs[PROPERTY_TEMPERATURE_SUPPLY],
-                outputs[PROPERTY_TEMPERATURE_RETURN],
-                outputs[PROPERTY_MASSFLOW],
-                self.pandapipes_net)
-        else:
-            raise ValueError("Simulation data not available.")
+        outputs[PROPERTY_VOLUMEFLOW] = mass_flow_to_volume_flow(
+            outputs[PROPERTY_MASSFLOW],
+            outputs[PROPERTY_TEMPERATURE_SUPPLY],
+            self.pandapipes_net)
+        outputs[PROPERTY_THERMAL_POWER] = mass_flow_and_temperature_to_heat_demand(
+            outputs[PROPERTY_TEMPERATURE_SUPPLY],
+            outputs[PROPERTY_TEMPERATURE_RETURN],
+            outputs[PROPERTY_MASSFLOW],
+            self.pandapipes_net)
 
         self.output.append(outputs)
```

Comment:
Is er een reden dat je hier de get_timeseries() weghaalt?

Reply:
Ja want die is geimplementeerd in de asset_abstract en hetzelfde voor alle assets, dus die moet niet pappart in assets staan